pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -48,39 +48,58 @@ func (d *DualWriterMode2) Mode() DualWriterMode {
 }
 
 // Create overrides the behavior of the generic DualWriter and writes to LegacyStorage and Storage.
-func (d *DualWriterMode2) Create(ctx context.Context, original runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
+func (d *DualWriterMode2) Create(ctx context.Context, in runtime.Object, createValidation rest.ValidateObjectFunc, options *metav1.CreateOptions) (runtime.Object, error) {
 	var method = "create"
 	log := d.Log.WithValues("method", method)
 	ctx = klog.NewContext(ctx, log)
 
+	accIn, err := meta.Accessor(in)
+	if err != nil {
+		return nil, err
+	}
+	if accIn.GetUID() != "" {
+		return nil, fmt.Errorf("UID should not be present: %v", accIn.GetUID())
+	}
+
+	if accIn.GetGenerateName() == "" {
+		return nil, fmt.Errorf("generate name is empty")
+	}
+
+	accIn.SetName(accIn.GetGenerateName())
```

Comment:
i think we might need something like this:
```suggestion
	if accIn.GetName() == "" && accIn.GetGenerateName() == "" {
		return nil, fmt.Errorf("name is empty")
	}

	if accIn.GetName() == "" && accIn.GetGenerateName() != "" {
	   accIn.SetName(names.SimpleNameGenerator.GenerateName(accIn.GetGenerateName()))
       accIn.SetGenerateName("")
	}
```